internal/ent/hooks/orgsetting_sso.go

```diff
@@ -133,3 +140,51 @@ func fallbackString(primary func() (string, bool), fallback func() (*string, err
 
 	return "", false
 }
+
+// disableEnforcementIfConfigChanged makes sure if any value changes in the idp config,
+// the enforcement is disabled. The "tested" value is turned off too so the user/org is forced
+// to test the connection before they can enable it again
+func disableEnforcementIfConfigChanged(ctx context.Context, m *generated.OrganizationSettingMutation) error {
+
+	if didSSOConfigChange(ctx, m) || m.Op() == ent.OpCreate {
```

Comment:
These shoudl be switched, so you can skip the more expensive check `didSSOConfigChange` when it is a createOp
```suggestion
	if  m.Op() == ent.OpCreate || didSSOConfigChange(ctx, m){
```